content/projects/gsoc/students.adoc

```diff
@@ -50,31 +81,205 @@ We highly recommend to make some contributions to the project while you work on
+  * For example, an introduction to subject matter experts.
+. Continue to discuss and plan the project with the community and the mentors:
+  * Work on the design document of the project.
+  ** Work on clarifying objectives and expectations,
+  ** Study, refine and discuss the design and the project plan,
+  ** Top-level architecture document:
+  *** Create diagrams of operation,
+  *** Answer questions such as "How is the user going to use this?", "What configurations are needed?", etc.,
+  *** Some people find it useful to write a mini user guide or how-to guide, as if the project was already done. This usually helps define the project.,
+  * Create an implementation plan with milestones per coding period.
+  * At this point it may be appropriate to discuss the project on the _jenkinsci-dev@googlegroups.com_ mailing list (link:https://groups.google.com/forum/#!forum/jenkinsci-dev[archives]). Talk to the mentor about it.
+. Setup your computer and your development environment to work on the project (see <<UsefulLinks>>).
+. Learn and discuss the process with the mentors:
+  * Setup the github project,
+  * Pull-requests,
+  * Code reviews.
+. Learn the coding style expectations:
+  * From the link:https://wiki.jenkins.io/display/JENKINS/Code+Style+Guidelines[Code Style Guidelines]
+  * By reading existing code
+  * When in doubt, ask mentors or submit a pull-request and ask for review
+  * The following are all important for readability:
+  ** Proper spacing,
+  ** Proper indentation,
+  ** Proper capitalization,
+  ** Proper class, member and method names.
+  * The following are important for functionality and quality:
+  ** Proper annotations,
+  ** Unit and integration testing in Jenkins.
+  * If your project is a plugin, you also need to learn the link:https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins[plugin release process].
+. We use link:https://issues.jenkins-ci.org/secure/Dashboard.jspa[Jira] to track GSoC tasks:
+  * Create an account using link:https://accounts.jenkins.io/login[this link].
+  * Become familiar with navigating Jira.
+
+== Coding periods
+
+Students are expected to...
+
+. Work on the GSoC project as it is a full-time employment.
+ * It means that 30..40 hours per week is an **expected** workload though it can be adjusted upon the agreements with mentors.
+ * Push code to github almost every day of every coding period.
+ * Chat a line or two about what you are doing, almost every coding day, in your project channel (writing code, writing tests, updating documentation, etc.).
+ ** Just saying "Hi, today I am working on these classes" or "writing tests for ..." is good enough, but you can of course interact more as needed.
+ * Write a short summary of the work done each week, published to:
+ ** A personal blog, or
+ ** To the Jenkins website blogs (see link:https://github.com/jenkins-infra/jenkins.io/blob/master/CONTRIBUTING.adoc#adding-a-blog-post[adding a blog post])
```

Comment:
or to the SIG mailing list.

Reply:
I also do not think that blogposting to jenkins.io each week is a good idea. Posting for milestones should be enough

Reply:
I agree.